back-end/src/routes/api.js

```diff
@@ -104,8 +98,203 @@ const uploadFile = (request, response) => {
   });
 };
 
-//uploadFileAsync(process.argv[2])
+/**
+ * Retrieve all photos uploaded by user.
+ * @param {email: String} request 
+ * @param {[{photo: Object, desc: String, uploadDate: String, title: String, likes: String}]} response 
+ */
+const getPhotoSocial = (request, response) => {
+  (async () => {
+    var statusCode = 400;
+    var message = "User does not exist";
+    const reqEmail = request.body.email;
+    console.debug(reqEmail);
+    
+    // For now this will be in the headers. Modify later maybe.
+    if (reqEmail) {
+      try {
+        console.debug("Querying photos");
+        
+        // Query for information based on input email. Use email index to search.
+        var queryParams = {
+          TableName: "Files",
+          IndexName: "email-file-index",
+          KeyConditionExpression: "#email = :email",
+          ExpressionAttributeNames: {
+            "#email": "email"
+          },
+          ExpressionAttributeValues: {
+            ":email": reqEmail
+          }
+        };
+        const queryData = await dynamodb.query(queryParams).promise();
+        const items = queryData.Items;
+
+        var responseData = [];
+        // Loop through photos to send all object data
+        prevTitle = "";
+        for (let i = 0; i < items.length; i++) {
+          if (items[i].title === prevTitle) {
+            continue;
+          }
+
+          // Get picture data from S3 to send in response.
+          var getParams = {
+            Bucket: BUCKET_NAME,
+            Key: items[i].file
+          };
+          var photoData = await s3.getObject(getParams).promise();
+          
+          responseObject = {
+            photo: photoData.Body.buffer,
+            title: items[i].title,
+            desc: items[i].desc,
+            likes: items[i].likes,
+            uploadDate: items[i].uploadDate
+          };
+
+          responseData.push(responseObject);
+          prevTitle = items[i].title;
+        }
+        statusCode = 200;
+        console.debug(responseData);
+        response.status(statusCode).send(responseData);
+      }
+      catch(e) {
+        console.debug("Could not retrieve photos", e);
+      }
+      console.debug("Retrieval successful");
+    }
+    else {
+      response.status(statusCode).send(message);
+    }
+  })().catch(e => {
+    console.debug("Something went wrong");
```

Comment:
same here, change to `error` and also include error `e` into the message

e.g.
```js
console.error(e);
```